x-pack/plugins/actions/server/builtin_action_types/xmatters.ts

```diff
@@ -35,26 +35,11 @@ export type ActionTypeConfigType = TypeOf<typeof ConfigSchema>;
 // secrets definition
 export type ActionTypeSecretsType = TypeOf<typeof SecretsSchema>;
 const secretSchemaProps = {
-  user: schema.nullable(schema.string()),
-  password: schema.nullable(schema.string()),
+  user: schema.maybe(schema.string()),
+  password: schema.maybe(schema.string()),
   secretsUrl: schema.maybe(schema.string()),
 };
-const SecretsSchema = schema.object(secretSchemaProps, {
-  validate: (secrets) => {
-    // user and password must be set together (or not at all)
-    // if user/password provided, then no secretsUrl should be present
-    if ((!secrets.password && secrets.user) || (secrets.password && !secrets.user)) {
-      return i18n.translate('xpack.actions.builtin.xmatters.invalidUsernamePassword', {
-        defaultMessage: 'Both user and password must be specified',
-      });
-    }
-    if (secrets.password && secrets.user && !secrets.secretsUrl) return;
-    if (!secrets.password && !secrets.user && secrets.secretsUrl) return;
-    return i18n.translate('xpack.actions.builtin.xmatters.invalidSecrets', {
-      defaultMessage: 'Either user and password or URL authentication must be specified',
-    });
-  },
-});
+const SecretsSchema = schema.object(secretSchemaProps);
```

Comment:
Moved this validation block to the `validateActionTypeSecrets` function below. Since we're using 

```
schema.object(secretSchemaProps, {
  validate: curry(validateActionTypeSecrets)(configurationUtilities),
})
```
the validation uses the `validateActionTypeSecrets` instead of the validation on the config schema.